internal/ethapi/api.go

```diff
@@ -502,6 +502,53 @@ func (s *PublicBlockChainAPI) GetBalance(ctx context.Context, address common.Add
 	return (*hexutil.Big)(state.GetBalance(address)), state.Error()
 }
 
+// GetProof returns the Merkle-proof for a given account and optionally some storage keys.
+func (s *PublicBlockChainAPI) GetProof(ctx context.Context, address common.Address, storageKeys []string, blockNr rpc.BlockNumber) (map[string]interface{}, error) {
+	state, _, err := s.b.StateAndHeaderByNumber(ctx, blockNr)
+	if state == nil || err != nil {
+		return nil, err
+	}
+
+	storageTrie := state.StorageTrie(address)
+	storageHash := types.EmptyRootHash
+	codeHash := state.GetCodeHash(address)
+	storageProof := make([]map[string]interface{}, len(storageKeys))
+
+	if storageTrie != nil {
+		storageHash = storageTrie.Hash()
+	} else {
+		codeHash = crypto.Keccak256Hash(nil)
+	}
+
+	for i := range storageKeys {
+		if storageTrie != nil {
+			storageProof[i] = map[string]interface{}{
+				"key":   storageKeys[i],
+				"value": state.GetState(address, common.HexToHash(storageKeys[i])),
+				"proof": common.ToHexArray(state.GetStorageProof(address, common.HexToHash(storageKeys[i]))),
+			}
+		} else {
+			storageProof[i] = map[string]interface{}{
+				"key":   storageKeys[i],
+				"value": common.Hash{},
+				"proof": []string{},
+			}
+		}
+	}
+
+	fields := map[string]interface{}{
```

Comment:
A more idiomatic way would be to define a type, and return that. 
Eg. 
```
type StorageResult struct{
   Address common.Address  `json:"address"`
 ...
```

Reply:
I agree, I created structs and returned them, which works nicly.